nv_peer_mem.c

```diff
@@ -290,9 +301,19 @@ static int nv_dma_map(struct sg_table *sg_head, void *context,
 		nv_mem_context->dma_mapping = dma_mapping;
 		nv_mem_context->sg_allocated = 1;
 		for_each_sg(sg_head->sgl, sg, nv_mem_context->npages, i) {
-			sg_set_page(sg, NULL, nv_mem_context->page_size, 0);
 			sg->dma_address = dma_mapping->dma_addresses[i];
-			sg->dma_length = nv_mem_context->page_size;
+			page_size = nv_mem_context->page_size;
+			if (i == nv_mem_context->npages - 1)
+				page_size = ((system_page_virt_end - system_page_virt_start +
```

Comment:
this can be either 4KB (x86) or 64KB (GPU BAR1) right?
a comment would be useful

Reply:
And don't forget on ppc64le, sometimes the system is using HUGE pages which is configurable (default 2MB on RHEL7/8, but should always be queried).